### PR TITLE
Bug/malformed descriptions

### PR DESCRIPTION
**Pull request by stephanvd**
See Bob's commit messages for clarification

### CODE REVIEW

**Review by lvonk — COMMENTED** [review]
Not sure if `m` does anything?

File: lib/mt940_structured/parsers/abnamro/transaction_parser.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def get_regex_for_line_61
 
     def enrich_transaction(transaction, line_86)
       transaction.contra_account = "NONREF" #default
-      line_86 = line_86.gsub(/:86:/, '')
+      line_86 = line_86.gsub(/:86:/m, '')
```
[the remark this answers is not in the document]
Comment (stephanvd) [reply lib/mt940_structured/parsers/abnamro/transaction_parser.rb:37]: line 86 can be multiline now, not really needed in this case but for consistency

File: lib/mt940_structured/parsers/sns/transaction_parser.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def get_regex_for_line_61
     def enrich_transaction(transaction, line_86)
       transaction.contra_account = "NONREF" unless transaction.contra_account
       line_86 = line_86.gsub(/:86:/, '')
```
[the remark this answers is not in the document]
Comment (stephanvd) [reply lib/mt940_structured/parsers/sns/transaction_parser.rb]: missed that one :)

File: spec/mt940_abnamro_spec.rb [hunk 3]
```diff
@@ -38,7 +38,7 @@
 
       context 'Description' do
         it 'have the correct description in case of a GIRO account' do
-          expect(@transaction.description).to eq('KPN - DIGITENNE    BETALINGSKENM.  000000042188659 5314606715                       BETREFT FACTUUR D.D. 20-05-2011 INCL. 1,44 BTW')
+          expect(@transaction.description).to eq('KPN - DIGITENNE    BETALINGSKENM.  0000000421886595314606715                       BETREFT FACTUUR D.D. 20-05-2011INCL. 1,44 BTW')
```
[the remark this answers is not in the document]
Comment (stephanvd) [reply spec/mt940_abnamro_spec.rb:41]: if i understand correctly we only add newlines for repeating :86: lines. not for blank follow up lines

[the remark this answers is not in the document]
Comment (lvonk) [reply spec/mt940_abnamro_spec.rb:41]: What I mean is shouldn't we have to keep the existing newline in this case (looking at the mt940 file)

File: lib/mt940_structured/file_content.rb [hunk 4]
```diff
@@ -2,9 +2,12 @@ class MT940Structured::FileContent
...
   R_EOF_ABN_AMRO = /^-$/
   R_EOF_TRIODOS = /^-$/
+  IBAN_LINE = /^\w{2}\d{2}\w{4}\d{6,}$/
+  BANK_NUMBER_LINE = /^\d{6,}$/
+  GIRO_NUMBER_LINE = /^P\d{6,}$/
 
-  def initialize(raw_lines, join_lines_by = ' ')
-    @raw_lines = raw_lines.map { |line| line.strip }
+  def initialize(raw_lines, join_lines_by = "\n")
+    @raw_lines = raw_lines.map(&:rstrip)
```
Comment (bforma) [inline lib/mt940_structured/file_content.rb:10]: I think we should avoid touching the original lines at all. That will only lead to differences with data from bank data providers.

**Review by bforma — CHANGES_REQUESTED** [review]

**Pull request merged**